### PR TITLE
ENH: Improve format option to CallbackPrintIteration, closes #858

### LINKED ISSUE #858: CallbackPrintIteration should accept a format string.
Currently `CallbackPrintIteration` has the following parameter style

```
Parameters
----------
text : string
    Text to display before the iteration count. Default: 'iter ='
```

which is later called like

```python
print("{} {}".format(self.text, self.iter))
```

We should change this to be similar to the format string used by `CallbackSaveToDisk` and `CallbackShow` which looks like

``` 
Parameters
----------
saveto : string
    Format string for the name of the file(s) where
    iterates are saved. The file name is generated as

        filename = saveto.format(cur_iter_num)

    where ``cur_iter_num`` is the current iteration number.
```

### PR DESCRIPTION
**Pull request by adler-j**

### CODE REVIEW

File: odl/solvers/util/callback.py [hunk 1]
```diff
@@ -234,26 +234,51 @@ class CallbackPrintIteration(SolverCallback):
...
+
+                print(fmt.format(cur_iter_num))
+
+            where ``cur_iter_num`` is the current iteration number.
         display_step : positive int, optional
             Number of iterations between output. Default: 1
+
+        Examples
+        --------
+        Create simple callback that prints iteration count
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: colon

File: odl/solvers/util/callback.py [hunk 2]
```diff
@@ -234,26 +234,51 @@ class CallbackPrintIteration(SolverCallback):
...
+
+        >>> callback = CallbackPrintIteration()
+        >>> callback(None)
+        iter = 0
+        >>> callback(None)
+        iter = 1
+
+        Create callback that prints iteration count every 2:th iterate with
+        a custom string.
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: colon

File: odl/solvers/util/callback.py [hunk 3]
```diff
@@ -234,26 +234,51 @@ class CallbackPrintIteration(SolverCallback):
...
+        --------
+        Create simple callback that prints iteration count
+
+        >>> callback = CallbackPrintIteration()
+        >>> callback(None)
+        iter = 0
+        >>> callback(None)
+        iter = 1
+
+        Create callback that prints iteration count every 2:th iterate with
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: 2nd

File: odl/solvers/util/callback.py [hunk 4]
```diff
@@ -263,8 +288,8 @@ def reset(self):
 
     def __repr__(self):
         """Return ``repr(self)``."""
-        textstr = '' if self.text == self._default_text else self.text
-        return 'CallbackPrintIteration({})'.format(textstr)
+        fmtstr = '' if self.fmt == self._default_fmt else self.fmt
+        return 'CallbackPrintIteration({})'.format(fmtstr)
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: This should include `display_step`. Probably simplest to use the helper:
```python
optargs = [('fmt', self.fmt, self._default_fmt),
           ('display_step', self.display_step, 1)]
inner_str = signature_string([], optargs)
return '{}({})'.format(self.__class__.__name__, inner_str)
```

**Comment by adler-j** [comment]
Merge after CI.

**Pull request merged**